homeassistant/components/hausbus/gateway.py

```diff
@@ -0,0 +1,170 @@
+"""Representation of a Haus-Bus gateway."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from pyhausbus.ABusFeature import ABusFeature
+from pyhausbus.BusDataMessage import BusDataMessage
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.Configuration import (
+    Configuration,
+)
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.ModuleId import ModuleId
+from pyhausbus.HomeServer import HomeServer
+from pyhausbus.IBusDataListener import IBusDataListener
+from pyhausbus.ObjectId import ObjectId
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+    
+from .const import DOMAIN
+from .cover import HausbusCover, Rollladen
+from .entity import HausbusEntity
+
+LOGGER = logging.getLogger(__name__)
+
+
+class HausbusGateway(IBusDataListener):
+    """Manages a Haus-Bus gateway."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
```

Comment:
use the extended config entry here. You might need to move the `type` to this file